### PR TITLE
Firebase app check

### PR DESCRIPTION
**Pull request by LendiHop**

### CODE REVIEW

File: src/firebase/use-firebase-app.hook.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import { isIOS } from '../config/system';
+
+export const useFirebaseApp = () =>
+  useEffect(() => {
+    (async () => {
+      if (firebase.apps.length === 0) {
+        await firebase.initializeApp({
+          projectId: 'templewallet-fa3b3',
+          appId: isIOS ? '1:1067475869467:ios:838d63298a5073892c3dbe' : '1:1067475869467:android:a65f23f3674ab6052c3dbe'
```
Comment (unixvb) [inline src/firebase/use-firebase-app.hook.ts:12]: move these values into a variables

File: src/navigator/main-stack.tsx [hunk 2]
```diff
@@ -65,9 +66,9 @@ export const MainStackScreen = () => {
   useAppLockTimer();
   useBeaconHandler();
 
-  const initAppCheck = () => void dispatch(checkApp.submit());
+  useFirebaseApp();
 
-  useTimerEffect(initAppCheck, APP_CHECK_INTERVAL);
+  useTimerEffect(() => dispatch(checkApp.submit()), APP_CHECK_INTERVAL);
```
Comment (unixvb) [inline src/navigator/main-stack.tsx:71]: move this useTimerEfect & APP_CHECK_INTERVAL into useFirebaseApp

File: src/store/security/security-epics.ts [hunk 3]
```diff
@@ -10,12 +11,26 @@ import { isIOS } from '../../config/system';
...
   action$.pipe(
     ofType(checkApp.submit),
-    switchMap(() =>
-      from(templeWalletApi.get<VersionsInterface>('mobile-check')).pipe(
+    switchMap(() => appCheck.activate('ignored', false)),
+    switchMap(() => appCheck.getToken()),
+    switchMap(appCheckToken =>
+      from(
+        templeWalletApi.get<appCheckPayload>('mobile-check', {
+          params: { platform: isIOS ? 'ios' : 'android', appCheckToken: appCheckToken.token }
```
Comment (unixvb) [inline src/store/security/security-epics.ts:28]: just use `Platform.OS` value

File: src/store/security/security-reducers.ts [hunk 4]
```diff
@@ -9,22 +9,25 @@ export const securityReducers = createReducer<SecurityState>(securityInitialStat
...
     ...state,
-    isForceUpdateNeeded: createEntity(payload)
+    isForceUpdateNeeded: createEntity(payload.isForceUpdateNeeded),
+    isAppCheckFailed: createEntity(payload.isAppCheckFailed)
   }));
   builder.addCase(checkApp.fail, (state, { payload }) => ({
     ...state,
-    isForceUpdateNeeded: createEntity(state.isForceUpdateNeeded.data, false, payload)
+    isForceUpdateNeeded: createEntity(state.isForceUpdateNeeded.data, false, payload),
+    isAppCheckFailed: createEntity(state.isAppCheckFailed.data, false, payload)
```
Comment (unixvb) [inline src/store/security/security-reducers.ts:31]: maybe let's set `isForceUpdateNeeded.data: false` & `isAppCheckFailed.data: false` is request failed?

File: src/store/security/security-reducers.ts [hunk 5]
```diff
@@ -9,22 +9,25 @@ export const securityReducers = createReducer<SecurityState>(securityInitialStat
   builder.addCase(enterPassword.fail, state => ({
     ...state,
     passwordAttempt: state.passwordAttempt + 1,
-    passwordLockTime: state.passwordAttempt >= MAX_PASSWORD_ATTEMPTS ? Date.now() : state.passwordLockTime
```
Comment (letier3110) [inline src/store/security/security-reducers.ts:12]: why

File: src/store/security/security-epics.ts [hunk 6]
```diff
@@ -10,12 +11,26 @@ import { isIOS } from '../../config/system';
...
 export const CheckAppEpic = (action$: Observable<Action>) =>
   action$.pipe(
     ofType(checkApp.submit),
-    switchMap(() =>
-      from(templeWalletApi.get<VersionsInterface>('mobile-check')).pipe(
+    switchMap(() => appCheck.activate('ignored', false)),
+    switchMap(() => appCheck.getToken()),
+    switchMap(appCheckToken =>
+      from(
+        templeWalletApi.get<appCheckPayload>('mobile-check', {
```
Comment (unixvb) [inline src/store/security/security-epics.ts:27]: split into 2 different env (stage tw api & prod tw api)

File: .env.dist [hunk 7]
```diff
@@ -1,5 +1,7 @@
 READ_ONLY_SIGNER_PUBLIC_KEY=edpkvWbk81uh1DEvdWKR4g1bjyTGhdu1mDvznPUFE2zDwNsLXrEb9K
 READ_ONLY_SIGNER_PUBLIC_KEY_HASH=tz1fVQangAfb9J1hRRMP2bSB6LvASD6KpY8A
+IOS_APP_ID=
+ANDROID_APP_ID=
```
Comment (unixvb) [inline .env.dist:4]: set default (public values)

File: .github/workflows/testflight-build.yml [hunk 8]
```diff
@@ -54,6 +54,14 @@ jobs:
...
+        run: |
+          cat ${{ secrets.FIREBASE_GOOGLE_SERVICE_ANDROID }} > android/app/google.services.json
+
+      - name: Write secrets to GoogleService.info.plist
+        run: |
+          cat ${{ secrets.FIREBASE_GOOGLE_SERVICE_IOS }} > ios/GoogleService.info.plist
+
       - name: Create .env file
         run: |
           cat << EOF > .env
```
Comment (unixvb) [inline .github/workflows/testflight-build.yml:67]: add IOS_APP_ID & ANDROID_APP_ID values here

File: src/firebase/use-firebase-app.hook.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+export const useFirebaseApp = () => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    (async () => {
+      if (firebase.apps.length === 0) {
+        await firebase.initializeApp({
+          projectId: 'templewallet-fa3b3',
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          appId: isIOS ? process.env.IOS_APP_ID! : process.env.ANDROID_APP_ID!
```
Comment (unixvb) [inline src/firebase/use-firebase-app.hook.ts:20]: reuse `getEnv` fn to omit ts-ignore / lint ignore

File: src/screens/app-check/app-check-warning.tsx [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+import { Divider } from '../../components/divider/divider';
+import { Icon } from '../../components/icon/icon';
+import { IconNameEnum } from '../../components/icon/icon-name.enum';
+import { InsetSubstitute } from '../../components/inset-substitute/inset-substitute';
+import { ScreenContainer } from '../../components/screen-container/screen-container';
+import { formatSize } from '../../styles/format-size';
+import { useColors } from '../../styles/use-colors';
+import { useAppCheckWarningStyles } from './app-check-warning.styles';
+
+const handleClose = () => RNExitApp.exitApp();
```
Comment (unixvb) [inline src/screens/app-check/app-check-warning.tsx:15]: we are not using handlers outside of the component. just use anonymous fn

**Review by unixvb — APPROVED** [review]

**Review by letier3110 — APPROVED** [review]

**Pull request merged**